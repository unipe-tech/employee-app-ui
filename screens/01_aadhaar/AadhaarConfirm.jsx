import { useEffect } from "react";
import { useDispatch } from "react-redux";
import { AppBar, Icon, IconButton } from "@react-native-material/core";
import { useNavigation } from "@react-navigation/core";
import { Alert, SafeAreaView, ScrollView } from "react-native";
import ProgressBarTop from "../../components/ProgressBarTop";
import { styles } from "../../styles";
import { MaterialIcons } from "react-native-vector-icons";
import { addCurrentScreen } from "../../store/slices/navigationSlice";
import Confirm from "../../apis/aadhaar/Confirm";

function AadhaarConfirm({ navigation }) {
  const dispatch = useDispatch();
  // const navigation = useNavigation();

  useEffect(() => {
    dispatch(addCurrentScreen("AadhaarConfirm"));
  }, []);

  const returnNull = () => null;

  const navigateToAadhaarVerify = () => navigation.navigate("AadhaarVerify");

  const backAlert = () => {
    Alert.alert(
      "Do you want to go back ?",
      "If you go back your AADHAAR Verification will have to be redone. Continue if you want to edit your Aadhaar number.",
      [
        { text: "No", onPress: returnNull, style: "cancel" },
        { text: "Yes", onPress: navigateToAadhaarVerify },
      ]
    );
  };

  return (
    <SafeAreaView style={styles.container}>
      <AppBar
        title="Aadhaar Data Confirmation"
        color="#4E46F1"
        leading={
          <IconButton
            testID="backIcon"
            icon={<MaterialIcons name="arrow-back" size={20} color="white" />}
            onPress={backAlert}
          />
        }
      />

      <ProgressBarTop step={1} />

      <ScrollView keyboardShouldPersistTaps="handled">
        <Confirm />
      </ScrollView>
    </SafeAreaView>
  );
}

export default AadhaarConfirm;
