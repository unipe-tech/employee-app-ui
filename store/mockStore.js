const store = {
  // Auth

  auth: {
    id: 1,
    phoneNumber: "9876543210",
    verifyStatus: "PENDING",
  },

  // Aadhaar

  aadhaar: {
    data: {
      aadhaar_data: {
        name: "Xyz garg",
      },
    },
    number: "789456123010",
    submitOTPtxnId: "987",
    verifyMsg: "This otp has been sent",
    verifyStatus: "PENDING",
  },

  // Bank

  bank: {
    accountNumber: "157894561230",
    ifsc: "INDB0000123",
    upi: "9876543210@upi",
    accountHolderName: "Xyz Goyal",
    verifyStatus: "PENDING",
    verifyMsg: "The verification is pending",
    bankName: "IndusInd Bank",
    bankBranch: "Mumbai",
    branchCity: "Mumbai",
  },

  // ESIC

  esic: {
    portal: { estCode: "", ipNumber: "" },
    familyDetails: {
      fatherHusband: { relation: "", name: "" },
      nominee: { relation: "", name: "" },
    },
    address: {
      present: { street: "", state: "", district: "", pincode: "" },
      permanent: { street: "", state: "", district: "", pincode: "" },
      nominee: { street: "", state: "", district: "", pincode: "" },
    },
  },

  // License

  license: {
    number: "",
    name: "",
    dob: "",
    bloodGroup: "",
    classes: {},
    rto: {},
    validity: {},
    verifyMsg: "",
    verifyStatus: "PENDING",
  },

  // Navigation

  navigation: { currentScreen: "Welcome" },

  // PAN

  pan: {
    dob: "02/02/2002",
    email: "xyz@unipe.co",
    gender: "Male",
    name: "Xyz Goyal",
    number: "ABCDE1234F",
    verifyMsg: "Verification Pending",
    verifyStatus: "PENDING",
  },

  // Profile

  profile: {
    maritalStatus: "Married",
    educationalQualification: "12th pass",
    alternatePhone: "+913216549870",
    email: "abc@unipe.co",
    selfie: "",
  },

  // Timer

  timer: { login: 2 * 60, aadhaar: 10 * 60 },
};

export default store;
