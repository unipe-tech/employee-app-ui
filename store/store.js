import { configureStore } from "@reduxjs/toolkit";
import authSlice from "./slices/authSlice";
import aadhaarSlice from "./slices/aadhaarSlice";
import panSlice from "./slices/panSlice";
import bankSlice from "./slices/bankSlice";
import esicSlice from "./slices/esicSlice";
import profileSlice from "./slices/profileSlice";
import navigationSlice from "./slices/navigationSlice";

export const store = configureStore({
  reducer: {
    auth: authSlice,
    aadhaar: aadhaarSlice,
    pan: panSlice,
    bank: bankSlice,
    profile: profileSlice,
    esic: esicSlice,
    navigation: navigationSlice
  },
});
