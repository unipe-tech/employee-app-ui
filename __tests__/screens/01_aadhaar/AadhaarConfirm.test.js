import React from "react";
import { render, fireEvent, waitFor, act } from "@testing-library/react-native";
import { NavigationContainer } from "@react-navigation/native";
import { Provider, useSelector } from "react-redux";
import { store } from "../../../store/store";
import { shallow } from "enzyme";
import { Button } from "@react-native-material/core";
import { Alert } from "react-native";
import spyOnAlert from "../../../testHelpers/spyOnAlert";
import AadhaarConfirm from "../../../screens/01_aadhaar/AadhaarConfirm";
import configureStore from "redux-mock-store";
import mockStore from "../../../store/mockStore";

const middleware = [];
const mockedStore = configureStore(middleware);
const data = {
  auth: {
    id: 1,
  },
  aadhaar: {
    data: {
      aadhaar_data: {
        name: "Tanish garg",
      },
    },
    number: "789456123010",
  },
};

const createdStore = mockedStore(mockStore);

const { pressAlertButton } = spyOnAlert();

describe("AadhaarConfirm Screen", () => {
  describe("UI Testing", () => {
    it("testing UI rendering", () => {
      const tree = render(
        <Provider store={createdStore}>
          <NavigationContainer>
            <AadhaarConfirm />
          </NavigationContainer>
        </Provider>
      );
      expect(tree).toMatchSnapshot();
    }, 8000);
  });
  describe("Testing Main Buttons", () => {
    it("testing No Button", () => {
      const navigate = jest.fn();
      const returnNull = jest.fn();
      const wrapper = render(
        <Provider store={createdStore}>
          <NavigationContainer>
            <AadhaarConfirm navigation={{ navigate }} />
          </NavigationContainer>
        </Provider>
      );
      act(async () => {
        await fireEvent.press(wrapper.getByText("No"));
        await expect(navigate).toHaveBeenCalledWith("AadhaarForm");
      });
    });
    it("testing Yes Button", () => {
      const navigate = jest.fn();
      const returnNull = jest.fn();
      const wrapper = render(
        <Provider store={createdStore}>
          <NavigationContainer>
            <AadhaarConfirm navigation={{ navigate }} />
          </NavigationContainer>
        </Provider>
      );
      act(async () => {
        await fireEvent.press(wrapper.getByText("Yes"));
        await expect(navigate).toHaveBeenCalledWith("PanForm");
      });
    });
  });
  describe("Back Button Testing", () => {
    it("NO returns null", () => {
      const returnNull = jest.fn();
      const navigate = jest.fn();
      const wrapper = render(
        <Provider store={createdStore}>
          <NavigationContainer>
            <AadhaarConfirm navigation={{ navigate }} />
          </NavigationContainer>
        </Provider>
      );
      fireEvent.press(wrapper.getByTestId("backIcon"));
      const spyAlert = jest
        .spyOn(Alert, "alert")
        //@ts-ignore
        .mockImplementation((title, message, callbackOrButtons) =>
          callbackOrButtons[1].onPress()
        );
      act(async () => {
        await fireEvent.press(wrapper.findByLabelText("No"));
        await pressAlertButton("Nothing happens");
        await expect(returnNull).toHaveBeenCalled();
        await expect(returnNull).toHaveReturned(null);
      }, 4000);
    });
  });

  it("YES navigates to OTP Screen", () => {
    const navigate = jest.fn();
    navigateToAadhaarVerify = jest.fn();
    const wrapper = render(
      <Provider store={createdStore}>
        <NavigationContainer>
          <AadhaarConfirm navigation={{ navigate }} />
        </NavigationContainer>
      </Provider>
    );
    fireEvent.press(wrapper.getByTestId("backIcon"));
    const spyAlert = jest
      .spyOn(Alert, "alert")
      //@ts-ignore
      .mockImplementation((title, message, callbackOrButtons) =>
        callbackOrButtons[1].onPress()
      );
    act(async () => {
      await fireEvent.press(wrapper.findByLabelText("Yes"));
      await pressAlertButton("Navigates to next screen");
      await expect(navigateToAadhaarVerify).toHaveBeenCalled();
      await expect(navigate).toHaveBeenCalledWith("AadhaarVerify");
    }, 4000);
  });
});
