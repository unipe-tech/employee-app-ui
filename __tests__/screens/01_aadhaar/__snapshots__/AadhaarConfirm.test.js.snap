// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AadhaarConfirm Screen UI Testing testing UI rendering 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "white",
      "height": "100%",
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "#4E46F1",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.23,
        "shadowRadius": 2.62,
        "zIndex": 4,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginEnd": 20,
              "marginStart": -12,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 24,
                "height": 48,
                "overflow": "hidden",
                "width": 48,
              },
              undefined,
            ]
          }
          testID="backIcon"
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onLayout={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
                  },
                ]
              }
            />
            <Text />
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 20,
                "fontWeight": "500",
                "letterSpacing": 0.15,
              },
              Object {
                "color": "#000000",
              },
              Array [
                Object {
                  "color": "#FFFFFF",
                },
                undefined,
              ],
            ]
          }
        >
          Aadhaar Data Confirmation
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(1,0,0,0)",
          },
          Object {
            "flexDirection": "column",
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(1,0,0,0)",
              "flexDirection": "row",
              "justifyContent": "space-around",
            },
            Object {
              "flexDirection": "row",
              "height": 40,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4aae4f",
                "borderColor": "#4aae4f",
                "borderRadius": 15,
                "borderWidth": 0,
                "height": 30,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": 30,
                "zIndex": 2,
              }
            }
          >
            <Text />
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#4aae4f",
                "borderRadius": 15,
                "borderWidth": 5,
                "height": 30,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": 30,
                "zIndex": 2,
              }
            }
          >
            <Text />
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#a4d4a5",
                "borderColor": "#4aae4f",
                "borderRadius": 15,
                "borderWidth": 0,
                "height": 30,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": 30,
                "zIndex": 2,
              }
            }
          >
            <Text />
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#a4d4a5",
                "borderColor": "#4aae4f",
                "borderRadius": 15,
                "borderWidth": 0,
                "height": 30,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": 30,
                "zIndex": 2,
              }
            }
          >
            <Text />
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#a4d4a5",
                "borderColor": "#4aae4f",
                "borderRadius": 15,
                "borderWidth": 0,
                "height": 30,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": 30,
                "zIndex": 2,
              }
            }
          >
            <Text />
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#a4d4a5",
                "borderColor": "#4aae4f",
                "borderRadius": 15,
                "borderWidth": 0,
                "height": 30,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": 30,
                "zIndex": 2,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "justifyContent": "space-around",
            },
            Object {
              "flexDirection": "row",
              "paddingVertical": 4,
            },
            Object {
              "alignItems": "center",
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 12,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                Object {
                  "color": "#000000",
                },
                Object {
                  "fontFamily": undefined,
                  "fontSize": 13,
                },
              ]
            }
          >
            Mobile Number
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 12,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                Object {
                  "color": "#4aae4f",
                },
                Object {
                  "fontFamily": undefined,
                  "fontSize": 13,
                },
              ]
            }
          >
            Aadhaar Card
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 12,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                Object {
                  "color": "#000000",
                },
                Object {
                  "fontFamily": undefined,
                  "fontSize": 13,
                },
              ]
            }
          >
            PAN Card
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 12,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                Object {
                  "color": "#000000",
                },
                Object {
                  "fontFamily": undefined,
                  "fontSize": 13,
                },
              ]
            }
          >
            Bank Account
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 12,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                Object {
                  "color": "#000000",
                },
                Object {
                  "fontFamily": undefined,
                  "fontSize": 13,
                },
              ]
            }
          >
            Profile
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 12,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                Object {
                  "color": "#000000",
                },
                Object {
                  "fontFamily": undefined,
                  "fontSize": 13,
                },
              ]
            }
          >
            Photo
          </Text>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    keyboardShouldPersistTaps="handled"
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "height": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "#230C45",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "marginRight": 20,
              "marginTop": 20,
            }
          }
        >
          Are these your AADHAAR details ?
          

        </Text>
        <Text
          style={
            Object {
              "color": "#230C45",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "marginLeft": 55,
              "marginTop": 10,
            }
          }
        >
          Number: 
          789456123010
        </Text>
        <Text
          style={
            Object {
              "color": "#230C45",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "marginLeft": 55,
              "marginTop": 10,
            }
          }
        >
          Name: 
          Xyz garg
        </Text>
        <Text
          style={
            Object {
              "color": "#230C45",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "marginLeft": 55,
              "marginTop": 10,
            }
          }
        >
          Date of Birth: 
        </Text>
        <Text
          style={
            Object {
              "color": "#230C45",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "marginLeft": 55,
              "marginTop": 10,
            }
          }
        >
          Gender: 
        </Text>
        <View
          style={
            Object {
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "#EB5757",
                "borderBottomEndRadius": 4,
                "borderBottomStartRadius": 4,
                "borderTopEndRadius": 4,
                "borderTopStartRadius": 4,
                "fontSize": 20,
                "height": 60,
                "marginTop": 20,
                "padding": 10,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 1.41,
                "width": 150,
              }
            }
            type="solid"
          >
            <View
              style={
                Array [
                  Object {
                    "borderBottomEndRadius": 4,
                    "borderBottomStartRadius": 4,
                    "borderTopEndRadius": 4,
                    "borderTopStartRadius": 4,
                    "overflow": "hidden",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onLayout={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 36,
                      "justifyContent": "center",
                      "minWidth": 64,
                      "paddingEnd": 16,
                      "paddingStart": 16,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
                      },
                    ]
                  }
                />
                <Text
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 1.25,
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "#000000",
                      },
                      Array [
                        Object {
                          "color": "#000000",
                        },
                        Object {
                          "opacity": 1,
                          "textTransform": "none",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  No
                </Text>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "#4E46F1",
                "borderBottomEndRadius": 4,
                "borderBottomStartRadius": 4,
                "borderTopEndRadius": 4,
                "borderTopStartRadius": 4,
                "fontSize": 20,
                "height": 60,
                "marginLeft": 40,
                "marginTop": 20,
                "padding": 10,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 1.41,
                "width": 150,
              }
            }
            testID="YesButton"
            type="solid"
          >
            <View
              style={
                Array [
                  Object {
                    "borderBottomEndRadius": 4,
                    "borderBottomStartRadius": 4,
                    "borderTopEndRadius": 4,
                    "borderTopStartRadius": 4,
                    "overflow": "hidden",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onLayout={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 36,
                      "justifyContent": "center",
                      "minWidth": 64,
                      "paddingEnd": 16,
                      "paddingStart": 16,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
                      },
                    ]
                  }
                />
                <Text
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 1.25,
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "#000000",
                      },
                      Array [
                        Object {
                          "color": "#FFFFFF",
                        },
                        Object {
                          "opacity": 1,
                          "textTransform": "none",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Yes
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "marginTop": 30,
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
